### ISSUE TITLE
Easy posts preset exposed in global settings (with solution)

### ISSUE THREAD

**Issue by jgladwill** [soflyy/oxygen-bugs-and-features#3534]
# TLDR Version
Change line 162 in `oxygen\component-framework\components\classes\easy-posts-templates\easy-posts-presets.php` to 
```php
"slug" => "masonry-image-w-rectangle-overlap",
```
It's a duplicate slug from line 150

---

**Sandbox with duplication preset**
https://oxygen-qnp9j3xjxxryl.oxygen-demo.qsandbox.org/wp-admin/admin.php?page=ct_export_import

**Sandbox with solution implemented via Plugin File Editor**
https://oxygen-qf7v7xfx9yryl.oxygen-demo.qsandbox.org/wp-admin/admin.php?page=ct_export_import
(See solution below)

## Steps to reproduce

1. Install oxygen@4.8.final_
2. Navigate to wp-admin/admin.php?page=ct_export_import
3. In the global settings export field, notice `"slug": "masonry-image-w-gradient-overlap",`, this should not be shown (see below)

In `oxygen\component-framework\admin\import-export.php`, line 140-156
```php
// remove defaults presets from export
$default_presets = apply_filters("oxygen_vsb_element_presets_defaults", array());

foreach ($default_presets as $element_name => $presets) {
	if (empty($element_presets[$element_name])) {
		continue;
	}

	foreach ($presets as $key => $preset) {
		$index = array_search($preset, $element_presets[$element_name]);
		if ($index !== false) {
			unset($element_presets[$element_name][$index]);
		}
	}
	// re-index array to keep JSON clean from indexes and not overwrite other presets
	$element_presets[$element_name] = array_values($element_presets[$element_name]);
}
```

In addition ...

4. Copy-paste the global settings from the export field into the import field and click "Import Global Settings"
5. This should produce the same global settings, but instead duplicates the preset because it's seen as a new preset altogether
6. Each time you import an exported global settings json without removing the exposed preset, it will create another duplicate with no way of removing it (since it should not be exposed in the first place)
7. Oxygen also accesses this file in-editor to save presets, which is likely causing #2471 

### Solution:
In `oxygen\component-framework\components\classes\easy-posts-templates\easy-posts-presets.php`, line 162

Note that `"slug" => "masonry-image-w-gradient-overlap",` occurs twice between lines 148 - 170

```php
array(
    "name" => __("Masonry - Image w/ Gradient Overlap","oxygen"),
    "slug" => "masonry-image-w-gradient-overlap",
    "options" => array(
        "original" => array(
            "code-php" => file_get_contents(OXYGEN_VSB_EASY_POSTS_TEMPLATES_PATH."masonry-image-w-gradient-overlap.php"),
            "code-css" => file_get_contents(OXYGEN_VSB_EASY_POSTS_TEMPLATES_PATH."masonry-image-w-gradient-overlap.css"),
            "title_size"=>"","title_size-unit"=>"px","title_color"=>"","title_hover_color"=>"","meta_size"=>"","meta_size-unit"=>"px","meta_color"=>"","content_size"=>"","content_size-unit"=>"px","content_color"=>"","read_more_display_as"=>"button","read_more_size"=>"0.8","read_more_size-unit"=>"em","read_more_text_color"=>"white","read_more_text_hover_color"=>"","read_more_button_color"=>"black","read_more_button_hover_color"=>"","paginate_color"=>"","paginate_alignment"=>"center","paginate_link_color"=>"","paginate_link_hover_color"=>"","posts_5050_below"=>"never","posts_100_below"=>"never",
        )
    )
),

array(
    "name" => __("Masonry - Image w/ Rectangle Overlap","oxygen"),
    "slug" => "masonry-image-w-gradient-overlap",
    "options" => array(
        "original" => array(
            "code-php" => file_get_contents(OXYGEN_VSB_EASY_POSTS_TEMPLATES_PATH."masonry-image-w-rectangle-overlap.php"),
            "code-css" => file_get_contents(OXYGEN_VSB_EASY_POSTS_TEMPLATES_PATH."masonry-image-w-rectangle-overlap.css"),
            "title_size"=>"","title_size-unit"=>"px","title_color"=>"","title_hover_color"=>"","meta_size"=>"","meta_size-unit"=>"px","meta_color"=>"","content_size"=>"","content_size-unit"=>"px","content_color"=>"","read_more_display_as"=>"button","read_more_size"=>"0.8","read_more_size-unit"=>"em","read_more_text_color"=>"white","read_more_text_hover_color"=>"","read_more_button_color"=>"black","read_more_button_hover_color"=>"","paginate_color"=>"","paginate_alignment"=>"center","paginate_link_color"=>"","paginate_link_hover_color"=>"","posts_5050_below"=>"never","posts_100_below"=>"never",
        )
    )
),
```

## Steps to fix
Note: This only works on a fresh install, since the oxygen preset bakes itself into the user's global settings export
1. Install a fresh copy of Wordpress with no plugins
2. Install oxygen@4.8.final_
3. Replace `oxygen\component-framework\components\classes\easy-posts-templates\easy-posts-presets.php` line 162 with `"slug" => "masonry-image-w-rectangle-overlap",`
4. Activate Oxygen plugin
5. Navigate to wp-admin/admin.php?page=ct_export_import
6. Notice the global settings export now does not include any default presets

**Comment by jgladwill** [comment]
### Update

I was able to confirm that this and #2471 are the same issue, saving a custom easy post preset causes duplicated default preset

The solution provided above successfully resolves both issues, however, once the duplicate presets are "baked in" to the global settings, they need to be removed by other means.

**Unresolved Sandbox**
https://oxygen-qf7v7xfx9yryl.oxygen-demo.qsandbox.org/?ct_template=test&ct_builder=true

1. Create a new Easy Posts preset using the "Save New Preset" dialog
2. The new preset will be added to the preset list along with another duplicate copy of "Masonry - Image w/ Rectangle Overlap"

**Comment by jgladwill** [comment]
![Before](https://i.imgur.com/I0bA6AE.jpeg)

![After](https://i.imgur.com/dfwMC7N.jpeg)

**Comment by Kpudlo** [comment]
Thank you for the detailed findings. We're going to close this as a duplicate in favor of https://github.com/soflyy/oxygen-bugs-and-features/issues/2471 and update our internal issue with this information.
